misc/helper-scripts/rosmsg_to_asn1.py

```diff
@@ -0,0 +1,417 @@
+        logging.error(
+            "Could not load the package {0}."
+            "Is it in your package path?".format(module_name))
+    return module
+
+
+def get_message_dict(module, msg):
+    """
+    Returns the dictionary representing a message type from a module.
+    :param module: The Python module contining the message.
+    :param msg: The message type.
+    """
+    msg_dict = None
+    obj = None
+    try:
+        # Create an object for introspection
+        obj = module.__dict__[msg]
+    except KeyError as e:
+        logging.error(
+            "Couldn't find message {0} in module {1}".format(msg, module))
+        return msg_dict
+    try:
+        msg_dict = dict(
+            type=obj._type, slots=obj.__slots__, slot_types=obj._slot_types,
+            full_text=obj._full_text)
+    except KeyError as e:
+        logging.error(
+            "Could not read the information of message {}.".format(msg))
+        return None
+    logging.debug("Dictionary for message {0} is:".format(msg))
+    logging.debug(pprint.pformat(msg_dict))
+    return msg_dict
+
+
+def find_field_type(slot_type):
+    """
+    Finds the field type of a field. If it is an array it returns the data type
+    of the array along the array type and its length.
+    :param slot_type: The type of the slot whose field type is to be found.
+    :return slot_type: The base data type of the slot (if it is an array).
+    :return field_type: The field_type of the slot.
+    :return length: The length of the array.
+    """
+    bracket = "["
+    length = 1
+    field_type = "simple"
+    idx = slot_type.find(bracket)
+    if idx == -1:
+        return slot_type, field_type, length
+    else:
+        if slot_type[idx + 1] == "]":
+            field_type = "variable-length"
+        else:
+            field_type = "fixed-length"
+            idx_end = slot_type.find("]")
+            str_length = slot_type[idx+1:idx_end]
+            length = int(str_length)
+        slot_type = slot_type[0:idx]
+    return slot_type, field_type, length
+
+
+def find_ASN1_type(slot_type):
+    """
+    Given the ROS type of a field finds and returns the equivalent ASN.1 type.
+    If the type is not primitive a flag is set so that a dependency is
+    generated.
+    :param slot_type: The ROS field type.
+    :return asn1_type: The ASN.1 equivalent type of the ROS type.
+    :return lib_asn: The library the ASN.1 type belongs to.
+    :return primitive: Flag to denote if the field type is primitive.
+    """
+    primitive = False
+    if slot_type in ros_primitive_type:
+        asn1_type = primitive_types_map[slot_type]
+        lib_asn = libraries[asn1_type]
+        primitive = True
+    else:
+        composed_type = slot_type.split("/")
+        if len(composed_type) == 1:
+            logging.warn("The length of the composed_type is 1!")
+        asn1_type = composed_type[1]
+        # primitive_types_map[slot_type] = asn1_type
+        lib_asn = composed_type[1]+'-Types'
```

Comment:
This looks like it might be bad when `len(composed_type) == 1`. Prefer to `return None, None, False` after the warning.

Reply:
Correct. It should fail with an error after that. Should I log it as an error better?